modules/db-writer/src/main/scala/fi/spectrumlabs/db/writer/classes/Handle.scala

```diff
@@ -82,17 +167,288 @@ object Handle {
+        case _                                      => false
       }
+
+    private def resolveDepositOrder(
+      deposit: Deposit,
+      input: TxInput,
+      tx: AppliedTransaction
+    ): Option[ExecutedDepositOrderInfo] = for {
+      userRewardOut <-
+        tx.txOutputs.find { txOut =>
+          txOut.fullTxOutValue.contains(deposit.coinLq) && checkForPubkey(
+            deposit.rewardPkh,
+            txOut.fullTxOutAddress.addressCredential
+          )
+        }
+      amountLq <- userRewardOut.fullTxOutValue.find(deposit.coinLq)
+      poolIn   <- tx.txInputs.filterNot(_.txInRef == input.txInRef).headOption
+      poolOut <- tx.txOutputs.find(
+        _.fullTxOutValue.contains(deposit.poolId)
+      )
+    } yield ExecutedDepositOrderInfo(
+      amountLq._2,
+      userRewardOut.fullTxOutRef,
+      poolIn.txInRef,
+      poolOut.fullTxOutRef,
+      tx.slotNo + cardanoConfig.startTimeInSeconds,
+      input.txInRef
+    )
+
+    private def resolveSwapOrder(swap: Swap, input: TxInput, tx: AppliedTransaction): Option[ExecutedSwapOrderInfo] =
+      for {
+        userRewardOut <- tx.txOutputs.find { txOut =>
+          txOut.fullTxOutValue.contains(swap.quote) && checkForPubkey(
+            swap.rewardPkh,
+            txOut.fullTxOutAddress.addressCredential
+          )
+        }
+        actualQuote <- userRewardOut.fullTxOutValue.find(swap.quote)
+        poolIn      <- tx.txInputs.filterNot(_.txInRef == input.txInRef).headOption
+        poolOut     <- tx.txOutputs.find(_.fullTxOutValue.contains(swap.poolId))
+      } yield ExecutedSwapOrderInfo(
+        actualQuote._2,
+        userRewardOut.fullTxOutRef,
+        poolIn.txInRef,
+        poolOut.fullTxOutRef,
+        tx.slotNo + cardanoConfig.startTimeInSeconds,
+        input.txInRef
+      )
+
+    private def resolveRedeemOrder(
+      redeem: Redeem,
+      input: TxInput,
+      tx: AppliedTransaction
+    ): Option[ExecutedRedeemOrderInfo] =
+      for {
+        userRewardOut <- tx.txOutputs.find { txOut =>
+          txOut.fullTxOutValue.contains(redeem.coinX) && txOut.fullTxOutValue.contains(
+            redeem.coinY
+          ) && checkForPubkey(
+            redeem.rewardPkh.getPubKeyHash,
+            txOut.fullTxOutAddress.addressCredential
+          )
+        }
+        actualX <- userRewardOut.fullTxOutValue.find(redeem.coinX)
+        actualY <- userRewardOut.fullTxOutValue.find(redeem.coinY)
+        poolIn  <- tx.txInputs.filterNot(_.txInRef == input.txInRef).headOption
+        poolOut <- tx.txOutputs.find(_.fullTxOutValue.contains(redeem.poolId))
+      } yield ExecutedRedeemOrderInfo(
+        actualX._2,
+        actualY._2,
+        userRewardOut.fullTxOutRef,
+        poolIn.txInRef,
+        poolOut.fullTxOutRef,
+        tx.slotNo + cardanoConfig.startTimeInSeconds,
+        input.txInRef
+      )
+
+    override def handle(in: NonEmptyList[TxEvent]): F[Unit] =
+      in.traverse {
+        case _: UnAppliedTransaction => ().pure[F]
```

Comment:
Shouldn't we process UnAppliedTransaction for any order in a specific way? Like - remove executed status and etc